kerchunk/grib2.py

```diff
@@ -811,7 +811,10 @@ def extract_datatree_chunk_index(
 
 
 def _map_grib_file_by_group(
-    fname: str, mapper: Optional[Callable] = None, storage_options=None
+    fname: str,
+    mapper: Optional[Callable] = None,
+    storage_options=None,
+    remote_options: Dict = None,
```

Comment:
Why is storage_options and remote_options different? We are only reading one remote file, no?

Aside: the type annotation should be `Optional[dict]` or `dict | None`.

Reply:
It is for the remote_options parameter in `grib_tree`
```
def grib_tree(
    message_groups: Iterable[Dict],
    remote_options=None,
) -> Dict:
```

Reply:
But why is it different

Reply:
Does `remote_options` and `target_options` come under the `storage_options`?

Reply:
I don't understand your question. You are trying to add both storage_options and remote_options to this function signature, and I don't see why they would be different - both reference the same remote grib file(s), no?

Reply:
I though it'd be confusing. So I made them different.  I'll make it `storage_options` across the code.

Reply:
I got confused with the signature of `grib_tree` function,
```
def grib_tree(
    message_groups: Iterable[Dict],
    remote_options=None,
   ```

Reply:
@martindurant  I've made the updates.